nevergrad/functions/images/imagelosses.py

```diff
@@ -3,38 +3,104 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+import cv2  # type: ignore
+import imquality.brisque as brisque  # type: ignore
+import lpips  # type: ignore
 import os
+import torch
 import typing as tp
 import numpy as np
 from nevergrad.functions.base import UnsupportedExperiment
+from nevergrad.common.decorators import Registry
+
+
+registry: Registry[tp.Any] = Registry()
 
 
 class ImageLoss:
     def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         pass
 
     def __call__(self, img: np.ndarray) -> float:
-        raise NotImplementedError
+        raise NotImplementedError(f"__call__ undefined in class {type(self)}")
 
 
-class SumAbsoluteDifferences(ImageLoss):
-    def __init__(self, reference: np.ndarray) -> None:
+class ImageLossWithReference(ImageLoss):
+    def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         if reference is None:
             raise ValueError("A reference is required")
         self.reference = reference
         super().__init__(reference)
+        assert len(self.reference.shape) == 3, self.reference.shape
+        assert self.reference.min() >= 0.0
+        assert self.reference.max() <= 256.0, f"Image max = {self.reference.max()}"
```

Comment:
isnt it 255?

Reply:
we meet values between 255 and 256, hence the change.